docs/hacking.md

```diff
@@ -147,3 +147,43 @@ Though not all parts of the codebase are currently formatted this way, it is
 recommended that when you submit a PR you run the respective formatters on the
 changed parts of the code, so that gradually the whole codebase is formatted
 uniformly.
+
+## Hacking on `build01`
+
+This section is for Tweagers only.
+
+First, set up your `build01` account according to the
+[handbook](https://github.com/tweag/meta/blob/master/handbook/eng/remote-builder.md).
+Don't forget to add the `groups = ["docker"]` line in your PR.
+
+Once the PR is merged, you can SSH into a NixOS non-privileged user. You can
+check out the `asterius` repo, set up your favorite text editor, make edits and
+push to the remote.
+
+To build/boot and run tests, a dev container needs to be built first. The
+`dev.rootless.Dockerfile` can be used to build an image which has the same UID
+with your user and doesn't mess up local file permissions:
+
+```sh
+$ docker build --build-arg UID=$(id -u) --file dev.rootless.Dockerfile --tag my_dev_image .
+```
```

Comment:
@TerrorJack when doing this I get the following:
```
[nix@build01:~/asterius]$ docker build --build-arg UID=$(id -u) --file dev.rootless.Dockerfile --tag my_dev_image .
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.40/build?buildargs=%7B%22UID%22%3A%221050%22%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=dev.rootless.Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&rm=1&session=h8ulmaggu56lovy7y5ejei36z&shmsize=0&t=my_dev_image&target=&ulimits=null&version=1": dial unix /var/run/docker.sock: connect: permission denied
```
Wonder why the permission issue :thinking:

Reply:
Why are you logged in as `nix` instead of `gkaracha`?

Reply:
I have no idea actually. I just used `ssh -i ~/.ssh/id-tweag-builder nix@build01.tweag.io`.

Reply:
Ah there are two tips:

* You should log in as `gkaracha@build01.tweag.io`
* Use [`mosh`](https://mosh.org/) to replace `ssh` to get a more stable remote shell

Reply:
> You should log in as gkaracha@build01.tweag.io

So true, how did I miss that :facepalm: I did that and now it seems to be working OK. It is supposed to take a very long time (because it does)?

Reply:
Worked just fine :ok_hand: